changelog/7942.improvement.rst

```diff
@@ -0,0 +1 @@
+Migrate from testdir to pytester in the acceptance_test.py file
```

Comment:
For internal improvements to our own testing suite, we don't add a changelog entry, because a user of the library isn't really affected by it.